src/fswAlgorithms/attGuidance/oneAxisSolarArrayPoint/oneAxisSolarArrayPoint.rst

```diff
@@ -0,0 +1,90 @@
+    * - transNavInMsg
+      - :ref:`NavTransMsgPayload`
+      - (optional) Input message containing the inertial position and velocity of the spacecraft. This message must be connected together with ``ephemerisInMsg`` to allow to compute :math:`{}^\mathcal{N}\hat{h}_\text{ref}`.
+    * - attRefOutMsg
+      - :ref:`AttRefMsgPayload`
+      - Output attitude reference message containing reference attitude, reference angular rates and accelerations.
+
+
+Detailed Module Description
+---------------------------
+A detailed mathematical derivation of the equations applied by this module can be found at `Calaon et al. <http://hanspeterschaub.info/Papers/Calaon2023.pdf>`__.
+The input parameter ``alignmentPriority`` allows to choose whether the first or the second constraint is strictly enforced. When ``alignmentPriority = 0``, the body heading :math:`{}^\mathcal{B}\hat{h}` and the inertial heading :math:`{}^\mathcal{N}\hat{h}_\text{ref}` match exactly, while the incidence angle on the solar arrays is as close to optimal as possible. On the contrary, when ``alignmentPriority = 1``, the solar array drive :math:`{}^\mathcal{B}\hat{a}_1` is perpendicular to the Sun direction, to ensure maximum power generation, while the body heading and the inertial heading are as close to parallel as possible.
+
+
+Module Assumptions and Limitations
+----------------------------------
+The limitations of this module are inherent to the geometry of the problem, which determines whether or not all the constraints can be satisfied. For example, as shown in  `Calaon et al. <http://hanspeterschaub.info/Papers/Calaon2023.pdf>`__, depending on the relative orientation of :math:`{}^\mathcal{B}h` and :math:`{}^\mathcal{B}a_1`, it may not be possible to  achieve perfect incidence angle on the solar arrays. Only when perfect incidence is obtained, it is possible to solve for the solution that also drives the body-fixed direction :math:`{}^\mathcal{B}a_2` close to the Sun. When perfect incidence is achievable, two solutions exist. If :math:`{}^\mathcal{B}a_2` is provided as input, this is used to determine which solution to pick. If this input is not provided, one of the two solution is chosen arbitrarily.
+
+Due to the difficulty in developing an analytical formulation for the reference angular rate and angular acceleration vectors, these are computed via second-order finite differences. At every time step, the current reference attitude and time stamp are stored in a module variable and used in the following time updates to compute angular rates and accelerations via finite differences.
+
+
+User Guide
+----------
+The required module configuration is::
```

Comment:
I'm confused by this info here.  You say these are required variables that must be set, but can't some of this be setup using optional input messages?

Reply:
This is true, and the comment next to the optional input specifies that. To be fair, these inputs aren't "optional", they need to be specified either as an input parameter or as an input message, but either one as to be provided. Would you like me to remove the optional inputs from the example module setup?

Reply:
Yes, please clarify this this.  You should list clearly which parameters can be either set directly or set via input message.  This will avoid confusion and having to read the code to see what is needed :-)

Reply:
I removed the inputs that are not strictly necessary from the example module setup. As far as specifying what information is required and in what form, I added an explanation in Detailed Module Description. Let me know what you think.